experiments/state/src/main/scala/org/apache/gearpump/streaming/state/user/api/PersistentTask.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.streaming.state.user.api
+
+import com.twitter.algebird.{Group, Monoid}
+import org.apache.gearpump.cluster.UserConfig
+import org.apache.gearpump.streaming.state.system.api.MonoidState
+import org.apache.gearpump.streaming.state.system.impl._
+import org.apache.gearpump.streaming.task.{Task, TaskContext}
+
+/**
+ *  User processor should extend this task to use PersistentState
+ */
+abstract class PersistentTask[T](taskContext: TaskContext, conf: UserConfig)
+  extends Task(taskContext, conf) {
+
+  private val stateConfig: PersistentStateConfig =
+    conf.getValue[PersistentStateConfig](PersistentStateConfig.NAME)
+      .getOrElse(throw new RuntimeException("PersistentStateConfig not found"))
+  private val checkpointInterval = stateConfig.getCheckpointInterval
+  private val checkpointStore = stateConfig.getCheckpointStoreFactory.getCheckpointStore(conf, taskContext)
+  private val checkpointManager = new CheckpointManager(checkpointInterval, checkpointStore)
+
+  /**
+   *  this is for user to get PersistentState
+   *  user should define or import an implicit algebird monoid
+   */
+  protected def persistentState(implicit monoid: Monoid[T]): MonoidState[T] =
```

Comment:
I think the return type should be PersistentState, instead of MonidState.

As long as user implement PersistentState's interface, user should be allowed to define its own PersistentState task.